app/models/folder.rb

```diff
@@ -43,6 +43,24 @@ def as_indexed_json(options={})
       indexes :is_deleted, type: "boolean"
       indexes :created_at, type: "date", format: "strict_date_optional_time||epoch_millis"
       indexes :updated_at, type: "date", format: "strict_date_optional_time||epoch_millis"
+
+      indexes :parent do
+        indexes :id, type: "string"
+        indexes :name, type: "string"
+      end
+
+      indexes :creator do
+        indexes :id, type: "string"
+        indexes :username, type: "string"
+        indexes :first_name, type: "string"
+        indexes :last_name, type: "string"
+        indexes :email, type: "string"
+      end
+
     end
   end
+
+  def creator
+    audits.find_by(action: "create").user
```

Comment:
A guard clause would be good incase audits.find_by returns nil. Though, maybe it'll throw a Record Not Found exception, which could be better/worse?  :\